code/game/jobs/job/support_chaplain.dm

```diff
@@ -36,134 +36,51 @@
-			switch(new_book_style)
-				if("Koran")
-					B.icon_state = "koran"
-					B.item_state = "koran"
-					for(var/area/chapel/main/A in world)
-						for(var/turf/T in A.contents)
-							if(T.icon_state == "carpetsymbol")
-								T.dir = 4
-				if("Scrapbook")
-					B.icon_state = "scrapbook"
-					B.item_state = "scrapbook"
-				if("Creeper")
-					B.icon_state = "creeper"
-					B.item_state = "syringe_kit"
-				if("White Bible")
-					B.icon_state = "white"
-					B.item_state = "syringe_kit"
-				if("Holy Light")
-					B.icon_state = "holylight"
-					B.item_state = "syringe_kit"
-				if("PlainRed")
-					B.icon_state = "athiest"
-					B.item_state = "syringe_kit"
-				if("Tome")
-					B.icon_state = "tome"
-					B.item_state = "syringe_kit"
-				if("The King in Yellow")
-					B.icon_state = "kingyellow"
-					B.item_state = "kingyellow"
-				if("Ithaqua")
-					B.icon_state = "ithaqua"
-					B.item_state = "ithaqua"
-				if("Scientology")
-					B.icon_state = "scientology"
-					B.item_state = "scientology"
-					for(var/area/chapel/main/A in world)
-						for(var/turf/T in A.contents)
-							if(T.icon_state == "carpetsymbol")
-								T.dir = 8
-				if("the bible melts")
-					B.icon_state = "melted"
-					B.item_state = "melted"
-				if("Necronomicon")
-					B.icon_state = "necronomicon"
-					B.item_state = "necronomicon"
-				if("Greentext")
-					B.icon_state = "greentext"
-					B.item_state = "greentext"
-				else
-					// if christian bible, revert to default
-					B.icon_state = "bible"
-					B.item_state = "bible"
-					for(var/area/chapel/main/A in world)
-						for(var/turf/T in A.contents)
-							if(T.icon_state == "carpetsymbol")
-								T.dir = 2
-
-			H.update_inv_l_hand() // so that it updates the bible's item_state in his hand
-
-			if(!B || !H.client)
-				break // prevents possible runtime errors
-			switch(input(H, "Look at your bible - is this what you want?") in list("Yes", "No"))
-				if("Yes")
-					accepted = 1
-				if("No")
-					if(outoftime)
-						to_chat(H, "Welp, out of time, buddy. You're stuck. Next time choose faster.")
-						accepted = 1
-
-		if(SSticker)
-			SSticker.Bible_icon_state = B.icon_state
-			SSticker.Bible_item_state = B.item_state
-			SSticker.Bible_name = B.name
-			SSticker.Bible_deity_name = B.deity_name
-		feedback_set_details("religion_deity", "[new_deity]")
-		feedback_set_details("religion_book", "[new_book_style]")
-
-
+	INVOKE_ASYNC(src, .proc/religion_pick, H)
+
+/datum/outfit/job/chaplain/proc/religion_pick(mob/living/carbon/human/user)
+	var/obj/item/storage/bible/B = new /obj/item/storage/bible(get_turf(user))
+	B.customisable = TRUE // Only the initial bible is customisable
+	user.put_in_l_hand(B)
+
+	var/religion_name = "Christianity"
+	var/new_religion = sanitize(copytext(input(user, "You are the Chaplain. What name do you give your beliefs? Default is Christianity.", "Name change", religion_name), 1, MAX_NAME_LEN))
```

Comment:
`clean_input` may work better here, this is a lot of nested procs.

Reply:
The `copytext()` is still needed I think, since that puts a maximum length on the name.